compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -326,6 +332,7 @@ impl Resolver<'_, '_> {
         let mut visitor = UnusedImportCheckVisitor {
             r: self,
             unused_imports: Default::default(),
+            single_unused_imports: Default::default(),
```

Comment:
What is the difference between `unused_imports` and `single_unused_imports`?
The previous version of the PR used `unused_imports` and seemed to work fine.

`unused_imports` is also still available at the end of this function when the redundancy checking is run.
Why cannot it be used? Due to borrow checker errors?
`unused_imports` can be moved out of `UnusedImportCheckVisitor` if necessary for redundancy checking, at that point it is no longer used for anything else.

Reply:
> What is the difference between `unused_imports` and `single_unused_imports`? The previous version of the PR used `unused_imports` and seemed to work fine.
> 
> `unused_imports` is also still available at the end of this function when the redundancy checking is run. Why cannot it be used? Due to borrow checker errors? `unused_imports` can be moved out of `UnusedImportCheckVisitor` if necessary for redundancy checking, at that point it is no longer used for anything else.

Hello, I defined a new all_unused_imports before and renamed it single_unused_imports. I think this name is relatively more accurate because it actually collects all single unused imports. unused_imports uses the root_id of the import in nested situations. It is inconvenient for me to quickly determine the duplication for each single import id.

Reply:
> What is the difference between `unused_imports` and `single_unused_imports`? The previous version of the PR used `unused_imports` and seemed to work fine.
> 
> `unused_imports` is also still available at the end of this function when the redundancy checking is run. Why cannot it be used? Due to borrow checker errors? `unused_imports` can be moved out of `UnusedImportCheckVisitor` if necessary for redundancy checking, at that point it is no longer used for anything else.

There is no need to use single_unused_imports. Has removed it.